src/transformers/models/rt_detr/configuration_rt_detr.py

```diff
@@ -0,0 +1,338 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" RT-DETR model configuration"""
+
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+from ..auto import CONFIG_MAPPING
+
+
+logger = logging.get_logger(__name__)
+
+RT_DETR_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "sbchoi/rtdetr_r50vd": "https://huggingface.co/sbchoi/rtdetr_r50vd/blob/main/config.json",
+}
+
+
+class RTDetrConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`RTDetrModel`]. It is used to instantiate a
+    RT-DETR model according to the specified arguments, defining the model architecture. Instantiating a configuration
+    with the defaults will yield a similar configuration to that of the RT-DETR
+    [checkpoing/todo](https://huggingface.co/checkpoing/todo) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        initializer_range (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        layer_norm_eps (`float`, *optional*, defaults to 1e-05):
+            The epsilon used by the layer normalization layers.
+        batch_norm_eps (`float`, *optional*, defaults to 1e-05):
+            The epsilon used by the batch normalization layers.
+        use_timm_backbone (`bool`, *optional*, defaults to `True`):
+            Whether the model use timm backbone.
+        backbone_config (`Union[Dict[str, Any], PretrainedConfig]`, *optional*):
+            The configuration of the backbone in a dictionary or the config object of the backbone.
+        backbone (`str`, *optional*, defaults to `"resnet50d"`):
+            Type of the backbone based on timm.
+        use_pretrained_backbone (`bool`, *optional*, defaults to `True`):
+            Whether to use pretrained weight for backbone model.
+        backbone_kwargs (`dict`, *optional*, defaults to `{'features_only': True, 'out_indices': [2, 3, 4]}`):
+            Keyword arguments to be passed to AutoBackbone when loading from a checkpoint
+            e.g. `{'out_indices': (0, 1, 2, 3)}`. Cannot be specified if `backbone_config` is set.
+        d_model (`int`, *optional*, defaults to 256):
+            Dimension of the layers.
+        encoder_in_channels (`list`, *optional*, defaults to `[512, 1024, 2048]`):
+            Multi level features input for encoder.
+        feat_strides (`List[int]`, *optional*, defaults to `[8, 16, 32]`):
+            Strides used in each feature map.
```

Comment:
Ideally those things can be inferred from the AutoBackbone API (cc @amyeroberts).

The Timm library for instance allows to get the in_channels and stride information for each of the stages: https://huggingface.co/docs/timm/en/feature_extraction#query-the-feature-information

Reply:
RTDETR only use timm library as backbone, there was conversion issue when I used AutoBackbone API let me recheck!

Reply:
@NielsRogge Found out that there are two section of using encoder_in_channels, so for the RTDETRModel usage I followed as your suggestion. However, in hybridencoder section it gets only configuration input so it was inevitable to remain this value.